limits.c

```diff
@@ -111,92 +110,96 @@ void limits_go_home(uint8_t cycle_mask)
 {
   if (sys.abort || !cycle_mask) { return; } // Block if system reset has been issued.
 
+  float target[N_AXIS];
+  float homing_rate;
+  // Determine travel distance to the furthest homing switch based on user max travel settings.
+  float min_seek_rate = 1e9; //arbitrary maximum=1km/s, will be reduced by axis setting below
+  float max_travel = 0;
   // Initialize homing in search mode to quickly engage the specified cycle_mask limit switches.
   uint8_t approach = ~0;  //approach has all bits set (negative dir) or none (positive)
-  float homing_rate;
   uint8_t idx;
-  uint8_t n_cycle = (2*N_HOMING_LOCATE_CYCLE+1);
-  float target[N_AXIS];
+  uint8_t n_cycle = (2*N_HOMING_LOCATE_CYCLE);
+  uint8_t n_active_axis = 0;
 
   uint8_t flipped = settings.homing_dir_mask>>X_DIRECTION_BIT;  //assumes keyme configuration.
-  //replace with an instance of `if (bitistrue(h_d_m,X_DIRECTION_BIT)) { flipped|=1<<X_AXIS;}` 
-  //for each axis if the bits line up differently
-
-  // Determine travel distance to the furthest homing switch based on user max travel settings.
-  float min_seek_rate = 1e9; //arbitrary maximum=1km/s, will be reduced by axis setting below
-  float max_travel = 0;
+  uint8_t axislock = 0; 
 
   for (idx=0; idx<N_AXIS; idx++){
     if (bit_istrue(cycle_mask,bit(idx))) {
+      n_active_axis++;
+      axislock |= (1<<(X_STEP_BIT+idx)); //assumes axes are in bit order.
```

Comment:
Where does axislock get used?  I don't see it anywhere... maybe hidden from view in the webpage diff?